packages/smooth_app/lib/pages/folksonomy/folksonomy_list.dart

```diff
@@ -0,0 +1,143 @@
+    _tags = Map<String, ProductTag>.from(widget.tags ?? {});
+  }
+
+  void _addTag(String key, String value, String? comment) {
+    // TODO: Make api call, verified authed
+    setState(() {
+      // _tags[key] = ProductTag(value: value, comment: comment);
+    });
+  }
+
+  void _editTag(
+      String oldKey, String newKey, String newValue, String? newComment) {
+    // TODO: Make api call, verified authed
+    setState(() {
+      // _tags.remove(oldKey);
+      // _tags[newKey] = ProductTag(value: newValue, comment: newComment);
+    });
+  }
+
+  void _deleteTag(String key) {
+    // TODO: Make api call, verified authed
+    setState(() {
+      _tags.remove(key);
+    });
+  }
+
+  void _showAddEditDialog(
+      {String? oldKey, String? oldValue, String? oldComment}) {
+    final TextEditingController keyController =
+        TextEditingController(text: oldKey);
+    final TextEditingController valueController =
+        TextEditingController(text: oldValue);
+    final TextEditingController commentController =
+        TextEditingController(text: oldComment);
+
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          // TODO: Localize
+          title: Text(oldKey == null ? 'Add Tag' : 'Edit Tag'),
+          content: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              TextField(
+                controller: keyController,
+                decoration: InputDecoration(labelText: 'Key'),
+              ),
+              TextField(
+                controller: valueController,
+                decoration: InputDecoration(labelText: 'Value'),
+              ),
+              TextField(
+                controller: commentController,
+                decoration: InputDecoration(labelText: 'Comment (optional)'),
+              ),
+            ],
+          ),
+          actions: [
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+              child: Text('Cancel'),
+            ),
+            TextButton(
+              onPressed: () {
+                if (oldKey == null) {
+                  _addTag(keyController.text, valueController.text,
+                      commentController.text);
+                } else {
+                  _editTag(oldKey, keyController.text, valueController.text,
+                      commentController.text);
+                }
+                Navigator.of(context).pop();
+              },
+              child: Text('Save'),
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        // TODO: Localize
+        title: Text('Product Tags'),
+      ),
+      body: ListView(
+        padding: const EdgeInsets.all(16.0),
+        children: _tags.entries.map((MapEntry<String, ProductTag> entry) {
+          return Dismissible(
```

Comment:
We try to avoid Dismissible, as nobody understands that there is an action behind

Reply:
Totally agree. I guess my question is how do you all want to handle delete and edit operations within a list? For now do we just want to add an edit and delete button inline for each list item? Do we want to have a detail view when clicking into a tag and deal with it there? Or a toggle in the AppBar or a FAB that allows deleting multiple and tapping on an item allows for editing? Kabob on each list item that shows a bottom sheet of actions? I guess my preference would be the latter. But not sure what design changes Pierre was talking about or if we want to make this consistent with other list views?

Reply:
probably not really. We'll have the feedback of the occasional panicked user who has added a wrong value, but delete can be done at a later time, or with a detaild bottom sheet as you suggest
What I would see as important is inline addition with an empty key and an empty value field to suggest you can contribute